libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/headline/PluggableHeadline.tsx

```diff
@@ -299,21 +301,47 @@ export class PluggableHeadline extends AbstractPluggableVisualization {
         }, []);
     }
 
-    private initializeProperties(visualizationProperties: IVisualizationProperties): void {
-        const controls = visualizationProperties?.controls;
-
-        const supportedProperties = getSupportedPropertiesControls(controls, this.supportedPropertiesList);
-        const initialProperties = {
-            supportedProperties: {
+    protected updateInstanceProperties(
+        options: IVisProps,
+        insight: IInsightDefinition,
+        insightPropertiesMeta: any,
+    ): void {
+        super.updateInstanceProperties(options, insight, insightPropertiesMeta);
+
+        const propertiesFromInsight = insightProperties(insight);
+        const primaryMeasuresCount = bucketsItems(insightBuckets(insight, BucketNames.MEASURES)).length;
+        const secondaryMeasuresCount = bucketsItems(
+            insightBuckets(insight, BucketNames.SECONDARY_MEASURES),
+        ).length;
+        const hasVisClassChanged = options.custom?.lastSavedVisClassUrl !== "local:headline";
+        const useDefaultMigrationProperties =
+            isInsight(insight) &&
+            insightId(insight) &&
+            !hasVisClassChanged &&
+            primaryMeasuresCount === 1 &&
+            secondaryMeasuresCount === 1;
+        if (this.settings?.enableNewHeadline && !propertiesFromInsight.controls?.comparison) {
+            const supportedProperties = getSupportedProperties(
+                propertiesFromInsight,
+                this.supportedPropertiesList,
+            );
+            const defaultProperties = useDefaultMigrationProperties
+                ? this.buildDefaultMigrationProperties()
+                : HEADLINE_DEFAULT_CONTROL_PROPERTIES;
+            this.visualizationProperties = {
                 controls: {
-                    ...supportedProperties,
-                    ...HEADLINE_DEFAULT_CONTROL_PROPERTIES,
+                    ...supportedProperties.controls,
+                    ...defaultProperties,
                 },
-            },
-        };
+            };
+        }
+    }
 
-        this.pushData({
-            initialProperties,
-        });
+    private buildDefaultMigrationProperties(): HeadlineControlProperties {
+        const controlProperties = HEADLINE_DEFAULT_MIGRATION_CONTROL_PROPERTIES;
+        controlProperties.comparison.labelConfig = {
```

Comment:
You are effectively modifying this imported constant. Is that what you really want? IMHO this should not be done like this and instead some cloning should be done first, ideally.

``` 
return {
   ...HEADLINE_DEFAULT_MIGRATION_CONTROL_PROPERTIES,
   comparison: {
       ...HEADLINE_DEFAULT_MIGRATION_CONTROL_PROPERTIES.comparison,
      labelConfig: { unconditionaValue: <translation> },
   },
};
```

Reply:
Fixed